### PR TITLE
mgr/dashboard: show a message to restart the rgw daemons after moving from single-site to multi-site

### PR DESCRIPTION
**Pull request by aaSharma14**
Fixes: https://tracker.ceph.com/issues/62984

Screenshot from 2023-09-26 18-00-11

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts [hunk 1]
```diff
@@ -386,6 +387,19 @@ export class RgwMultisiteDetailsComponent implements OnDestroy, OnInit {
...
     this.getDisableMigrate();
+    this.rgwDaemonService.list().subscribe((data: any) => {
+      const zonegroupNames = data.map((item: { [x: string]: any }) => item['zonegroup_name']);
+      const zoneNames = data.map((item: { [x: string]: any }) => item['zone_name']);
+      if (
+        this.defaultRealmId != '' &&
+        this.defaultZonegroupId != '' &&
+        this.defaultZoneId != '' &&
+        zonegroupNames.includes('default') &&
+        zoneNames.includes('default')
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts:398]: Can the name still be default and default after migrate?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html [hunk 2]
```diff
@@ -9,6 +9,14 @@
            (click)="enableRgwModule()">
            Enable RGW Module</a>
       </cd-alert-panel>
+      <cd-alert-panel   *ngIf="restartGatewayMessage"
+                        type="warning"
+                        spacingClass="mb-3"
+                        i18n>Please restart all Ceph Object Gateway instances in all zones to ensure consistent multisite configuration updates.
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html:15]:
```suggestion
                        i18n>Please restart all participating Ceph Object Gateway instances in default zone to ensure consistent while doing Import and Export
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html [hunk 3]
```diff
@@ -9,6 +9,14 @@
...
+        <a class="text-decoration-underline"
+           [routerLink]="'/services'">
+           Cluster->Services</a>
+      </cd-alert-panel>
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html:19]:
```suggestion
        <a class="text-decoration-underline"
           [routerLink]="'/services'">
           Services</a>
      </cd-alert-panel>
```

Comment (cloudbehl) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html:19]: Is it possible to filter rgw services?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts (hunk 1 above)
Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts]: Updated the PR

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html (hunk 2 above)
Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html:15]: Daemon restart is necessary after migrate to multisite or if a user created a new realm/zg/zone on fresh deployment..so decided to go on with a general message

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html [hunk 4]
```diff
@@ -9,6 +9,14 @@
            (click)="enableRgwModule()">
            Enable RGW Module</a>
       </cd-alert-panel>
+      <cd-alert-panel   *ngIf="restartGatewayMessage"
+                        type="warning"
+                        spacingClass="mb-3"
+                        i18n>Please restart all Ceph Object Gateway instances in all zones to ensure consistent multisite configuration updates.
+        <a class="text-decoration-underline"
+           [routerLink]="'/services'">
+           Cluster->Services</a>
```
[the remark this answers is not in the document]
Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html:18]: cd-alert-panel already has i18n attribute

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html [hunk 5]
```diff
@@ -9,6 +9,14 @@
            (click)="enableRgwModule()">
            Enable RGW Module</a>
       </cd-alert-panel>
+      <cd-alert-panel   *ngIf="restartGatewayMessage"
+                        type="warning"
+                        spacingClass="mb-3"
+                        i18n>Please restart all Ceph Object Gateway instances in all zones to ensure consistent multisite configuration updates.
+        <a class="text-decoration-underline"
+           routerLink='/services'>
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.html:17]:
```suggestion
           routerLink="/services">
```

**Review by nizamial09 — APPROVED** [review]

**Review by cloudbehl — APPROVED** [review]

**Pull request merged**